internal/core/cloud/metadata_test.go

```diff
@@ -393,7 +390,7 @@ func (suite *CloudMetadataTestSuite) TestNewInstanceNutanix() {
 		dmidecodeNutanix(), nil,
 	)
 
-	c, err := NewCloudInstance(mockCommand)
+	c, err := cloud.NewCloudInstance(mockCommand, nil)
```

Comment:
I have mixed feelings about passing a `nil` client in these tests where we know client is not needed. The `NewCloudInstance` API looks a bit inconsistent and leaking a bit the internals.

What about passing an instance of a client and expecting that no requests on that client would be executed?

Reply:
Okey!

Reply:
Done